src/main/java/com/github/netelli/ProductsDAO.java

```diff
@@ -46,6 +49,7 @@ public void deleteData() throws SQLException {
 
                 products.add(product);
             }
+            products.sort(Comparator.comparing(Product::getId));
```

Comment:
What it's doing here? You don't need to sort this data here, it's not required by main logic at all. **It's needed only for unit testing - so implement it in test src only.**